File: examples/lrmptest.go
package main

import (
	"bufio"
	"fmt"
	"github.com/robaho/lrmp"
	"log"
	"os"
)

type handler struct {
}

func (handler) ProcessData(p *lrmp.Packet) {
	fmt.Println("got a packet", string(p.GetDataBuffer()))
}

func (handler) ProcessEvent(event int, data interface{}) {
	fmt.Println("got an event", event, data)
}

func main() {
	fmt.Println("type a message and press 'enter' to send")

	profile := lrmp.NewProfile()
	profile.Handler = new(handler)

	l, err := lrmp.NewLrmp("225.0.0.100", 6000, 0, "en0", *profile)
	if err != nil {
		log.Fatal(err)
	}
	l.Start()

	scanner := bufio.NewScanner(os.Stdin)

	for scanner.Scan() {
		s := scanner.Text()

		bytes := []byte(s)

		p := lrmp.NewPacket(true, len(bytes))
		copy(p.GetDataBuffer(), bytes)
		p.SetDataLength(len(bytes))

		fmt.Println("sending message 100 times")

		for i := 0; i < 100; i++ {
			l.Send(p)
		}
	}

	l.Stop()

}

File: java/test/LrmpTest.java
import inria.net.lrmp.*;
import inria.util.Logger;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.Reader;

public class LrmpTest {
    public static void main(String[] args) throws LrmpException, IOException {
        Logger.setDebug();

        LrmpProfile profile = new LrmpProfile();
        profile.setEventHandler(new LrmpEventHandler() {
            @Override
            public void processData(LrmpPacket pack) {
                System.out.println("I received a packet, with "+
                        new String(pack.getDataBuffer(),pack.getOffset(),pack.getDataLength()));
            }

            @Override
            public void processEvent(int event, Object data) {
                System.out.println("got an event "+data);
            }
        });
        Lrmp lrmp = new Lrmp("225.0.0.100",6000,0,"en0",profile);
        lrmp.start();

        Reader r = new BufferedReader(new InputStreamReader(System.in));
        String s;

        while ((s = ((BufferedReader) r).readLine())!=null) {
            byte[] bytes = s.getBytes();
            LrmpPacket p = new LrmpPacket(true,bytes.length);
            System.arraycopy(bytes,0,p.getDataBuffer(),p.getOffset(),bytes.length);
            p.setDataLength(bytes.length);
            System.out.println("sending packet");
            for(int i =0;i<100;i++){
                lrmp.send(p);
            }
        }
    }
}
